### PR TITLE
[Write-stall] Add e2e tests - 2

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
- E2e tests for with default(10s) chunk-transfer-timeout and finite chunk-transfer-timeout.(5s)
- This test verifies that write operations stall for the expected duration when a write stall is induced while uploading the first chunk. It creates a file, writes data to it, and then calls Sync() to ensure the data is written to GCS. The test measures the time taken for the Sync() operation and asserts that it is less than < (Number-of-stalls*chunkTransferTimeout)+(2-5s to upload rest of the chunks)
- Updated README to add new tests.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - Automated

### CODE REVIEW

File: tools/integration_tests/emulator_tests/write_stall_test.go [hunk 1]
```diff
@@ -88,3 +88,131 @@ func TestChunkTransferTimeoutInfinity(t *testing.T) {
...
+}
+
+////////////////////////////////////////////////////////////////////////
+// Test Function (Runs once before all tests)
+////////////////////////////////////////////////////////////////////////
+
+func TestFiniteChunkTransferTimeout(t *testing.T) {
+	ts := &finiteChunkTransferTimeout{}
+	// Define flag set to run the tests.
+	flagsSet := [][]string{
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply tools/integration_tests/emulator_tests/write_stall_test.go]: Hmm.. Good idea. Done

File: tools/integration_tests/emulator_tests/write_stall_test.go [hunk 2]
```diff
@@ -88,3 +88,90 @@ func TestChunkTransferTimeoutInfinity(t *testing.T) {
...
+
+// Teardown for each test case.
+func (s *chunkTransferTimeoutSuite) Teardown(t *testing.T) {
+	setup.UnmountGCSFuse(rootDir)
+	assert.NoError(t, emulator_tests.KillProxyServerProcess(port))
+}
+
+// Test case: single write stall.
+func (s *chunkTransferTimeoutSuite) TestHandlesWriteStalls(t *testing.T) {
+	configPath := "./proxy_server/configs/write_stall_40s.yaml"
```
Comment (vadlakondaswetha) [inline tools/integration_tests/emulator_tests/write_stall_test.go:111]: code is duplicated . can you use t.run here also?

Comment (Tulsishah) [reply tools/integration_tests/emulator_tests/write_stall_test.go:111]: This configuration file related to the test itself, not to flag sets. For example, it enforces a one stall, and both flag sets execute the test with this configuration.

Comment (Tulsishah) [reply tools/integration_tests/emulator_tests/write_stall_test.go]: Updated as discussed offline.

**Review by vadlakondaswetha — APPROVED** [review]

**Pull request merged**